jupyter_collaboration/handlers.py

```diff
@@ -284,6 +286,8 @@ def on_close(self) -> None:
             # keep the document for a while in case someone reconnects
             self.log.info("Cleaning room: %s", self._room_id)
             self.room.cleaner = asyncio.create_task(self._clean_room())
+        if isinstance(self.room, DocumentRoom):
```

Comment:
Why not emitting the event for all room types?

Reply:
+1

Reply:
For YdocWebsocketHandler that handle global awareness websocket, its room type will be TransientRoom. I found its room id is fixed, called "JupyterLab:globalAwareness" and its creating and closing does not indicate any user joining or leaving the collaboration on a document. Hence we only emit when a websocket close where its room type is DocumentRoom.

Reply:
Global awareness being a `TransientRoom` doesn't mean that clients leaving a document in a `TransientRoom` should not emit an event. Maybe the event should not be emitted specifically when the room ID is `JupyterLab:globalAwareness`? This room ID is a bit unfortunate, as it should probably be a standard in the Jupyter ecosystem and not particularly for JupyterLab, so a better would have been `Jupyter:globalAwareness`.
@Zsailer do you think this room ID should be specified somewhere in the Jupyter schemas, since it's probably a convention we want to rely on?

Reply:
We would need to rely on RoomId to identity which document that the host has left from and close all the guest websockets associated as clients in that DocumentRoom. For a TransientRoom it seems that it is not tied to a specific document and mainly used by global awareness. We only want to emit an event when open and on_close has been successful as well.

Reply:
All I'm saying is that you could do something like that:
```suggestion
        if self.room.room_id != "JupyterLab:globalAwareness":
```
A `TransientRoom` is not just for global awareness, it could be for any other collaborative document that doesn't need to be persisted, and you probably want to emit join/leave events for them too.

Reply:
Sure.